compute_tools/src/http/api.rs

```diff
@@ -12,6 +15,44 @@ use serde_json;
 use tracing::{error, info};
 use tracing_utils::http::OtelName;
 
+async fn handle_spec_request(req: Request<Body>, compute: &Arc<ComputeNode>) -> Result<(), (String, StatusCode)> {
```

Comment:
Refactored this in a similar way too, thanks. Also tweaked waiting part to spawn a separate blocking thread with `tokio::task::spawn_blocking` to do not block main pool of worker threads and serve status requests. I didn't expect it to be a problem, but it seems to be a safer approach